pkg/storaged/content-views.jsx

```diff
@@ -862,12 +925,99 @@ function create_logical_volume(client, vgroup) {
+        },
+        {
+            value: "raid0",
+            title: _("Striped (RAID 0, at least two physical volumes)"),
+            min_pvs: 2,
+        },
+        {
+            value: "raid1",
+            title: _("Mirrored (RAID 1, at least two physical volumes)"),
+            min_pvs: 2,
+        },
+        {
+            value: "raid10",
+            title: _("Striped and mirrored (RAID 10, at least four physical volumes, even number)"),
+            min_pvs: 4,
+        },
+        {
+            value: "raid5",
+            title: _("Distributed parity (RAID 5, at least three physical volumes)"),
+            min_pvs: 3,
+        },
+        {
+            value: "raid6",
+            title: _("Double distributed parity (RAID 6, at least five physical volumes)"),
+            min_pvs: 5,
+        }
+    ];
+
     const vdo_package = client.get_config("vdo_package", null);
     const need_vdo_install = vdo_package && !(client.features.lvm_create_vdo || client.features.legacy_vdo);
 
     if (client.features.lvm_create_vdo || client.features.legacy_vdo || vdo_package)
         purposes.push({ value: "vdo", title: _("VDO filesystem volume (compression/deduplication)") });
 
+    const pvs_as_spaces = client.vgroups_pvols[vgroup.path].filter(pvol => pvol.FreeSize > 0).map(pvol => {
+        const block = client.blocks[pvol.path];
+        return { type: 'block', block, size: pvol.FreeSize, desc: "", pvol };
+    });
+
+    /* For layouts with redundancy, CreatePlainVolumeWithLayout will
+     * create as many subvolumes as there are selected PVs.  This has
+     * the nice effect of making the calculation of the maximum size of
+     * such a volume trivial.
+     */
+
+    function max_size(vals) {
+        const layout = vals.layout;
+        const pvs = vals.pvs.map(s => s.pvol);
+        const n_pvs = pvs.length;
+        const sum = pvs.reduce((sum, pv) => sum + pv.FreeSize, 0);
+        const min = Math.min.apply(null, pvs.map(pv => pv.FreeSize));
+
+        function metasize(datasize) {
+            const default_regionsize = 2 * 1024 * 1024;
+            const regions = Math.ceil(datasize / default_regionsize);
+            const bytes = 2 * 4096 + Math.ceil(regions / 8);
+            return vgroup.ExtentSize * Math.ceil(bytes / vgroup.ExtentSize);
+        }
+
+        if (layout == "linear") {
+            return sum;
+        } else if (layout == "raid0" && n_pvs >= 2) {
+            return n_pvs * min;
+        } else if (layout == "raid1" && n_pvs >= 2) {
+            return min - metasize(min);
+        } else if (layout == "raid10" && n_pvs >= 4) {
+            return Math.floor(n_pvs / 2) * (min - metasize(min));
+        } else if ((layout == "raid4" || layout == "raid5") && n_pvs >= 3) {
+            return (n_pvs - 1) * (min - metasize(min));
+        } else if (layout == "raid6" && n_pvs >= 5) {
+            return (n_pvs - 2) * (min - metasize(min));
+        } else
+            return 0;
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-17226-20230912-094641-c7f396be-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)